### PR TITLE
Adds custom error logging

### PR DESCRIPTION
**Pull request by colestrode**
Thanks for this library! We've added it to a few repos and it's been really helpful!

One feature that would be nice for our team would be the ability to add a custom logger. We have some standard logging formats that capture the timestamp and other information. It would be great if this handler would give us a way to use our own format.

I made this PR to add a custom logger to the library and updated the README with the usage. Let me know what you think!

Thanks again!

### CODE REVIEW

File: make-promises-safe.js [hunk 1]
```diff
@@ -3,11 +3,13 @@
 const event = 'unhandledRejection'
 
 process.on(event, function (err) {
-  console.error(err)
+  module.exports.logger.error(err)
```
Comment (mcollina) [inline make-promises-safe.js]: I would prefer us to just wrap this as a pure function, not as a `logger` object.

Comment (colestrode) [reply make-promises-safe.js]: so something like this?
```
module.exports.logError = function (err) { console.log(err); }
```

Comment (mcollina) [reply make-promises-safe.js]: I would go with `module.exports.logError = console.error`.

**Comment by colestrode** [comment]
I'm having a little trouble fitting a test into this existing framework. Normally I would use some sinon stubs to inspect things like this, would that be alright?

**Comment by mcollina** [comment]
you can use [proxyquire](http://npm.im/proxyquire)

**Comment by colestrode** [comment]
Not sure proxyquire would help in this case? there's no dependencies. Or maybe I'm missing something?

**Comment by davidmarkclements** [comment]
1. make a test fixture file  that exports a logError function 
2. copy paste the current test
3. require.resolve to the fixture (same approach as with main)
4. spawn fixture instead of main 
5. check that log output corresponds to the logError function output

**Comment by colestrode** [comment]
I think I got it! Thanks for your help :) I added a test and have it passing. I also added the `--fix` option to standard, this made it a lot easier for me to get things into shape and contribute.

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "Crash or abort if you get an unhandledRejection or multipleResolves",
   "main": "make-promises-safe.js",
   "scripts": {
-    "test": "standard | snazzy && tap test.js"
+    "test": "standard --fix | snazzy && tap test.js"
```
Comment (mcollina) [inline package.json]: Can you revert this? If you want, please put it in a separate script. We run `test` before any commit and CI.

File: test.js [hunk 3]
```diff
@@ -23,3 +23,16 @@ test('crashes the process on unhandled rejection', (t) => {
     t.is(code, 1)
   })
 })
+
+test('logs error', (t) => {
+  t.plan(1)
+
+  const child = spawn(process.execPath, ['-r', main, '-e', 'require("./make-promises-safe").logError = console.log; Promise.reject(new Error(\'kaboom\'))'])
```
Comment (mcollina) [inline test.js]: Can you customize the `logError` function somehow, maybe prefixing a string?

File: package.json (hunk 2 above)
Comment (colestrode) [reply package.json]: reverted

**Pull request merged**